fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -0,0 +1,151 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.persistence.ocfl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.vocabulary.DC;
+import org.apache.jena.vocabulary.RDF;
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.kernel.api.operations.RdfSourceOperation;
+import org.fcrepo.kernel.api.rdf.DefaultRdfStream;
+import org.fcrepo.persistence.api.CommitOption;
+import org.fcrepo.persistence.ocfl.api.OCFLObjectSessionFactory;
+import org.fcrepo.persistence.ocfl.impl.DefaultOCFLObjectSessionFactory;
+import org.fcrepo.persistence.ocfl.impl.FedoraOCFLMapping;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.stream.Stream;
+
+import static org.apache.jena.graph.NodeFactory.createLiteral;
+import static org.apache.jena.graph.NodeFactory.createURI;
+import static org.fcrepo.kernel.api.RdfLexicon.RDF_SOURCE;
+import static org.fcrepo.kernel.api.operations.ResourceOperationType.CREATE;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for {@link OCFLPersistentStorageSession}
+ *
+ * @author dbernstein
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class OCFLPersistentStorageSessionTest {
+
+
+    private OCFLPersistentStorageSession session;
+
+    @Mock
+    private FedoraToOCFLObjectIndex index;
+
+    @Mock
+    private FedoraOCFLMapping mapping;
+
+    private OCFLObjectSessionFactory objectSessionFactory;
+
+    private String parentId = "info:fedora/parent";
+    private String resourceId = parentId + "/child1";
+
+    @Mock
+    private RdfSourceOperation rdfSourceOperation;
+    @Rule
+    public TemporaryFolder tempFolder = new TemporaryFolder();
+
+    @Before
+    public void setUp() throws IOException {
+        final var stagingDir = tempFolder.newFolder("ocfl-staging");
+        final var repoDir = tempFolder.newFolder("ocfl-repo");
+        final var workDir = tempFolder.newFolder("ocfl-work");
+
+        this.objectSessionFactory = new DefaultOCFLObjectSessionFactory(stagingDir, repoDir, workDir);
+        session = createSession(index, objectSessionFactory);
+
+    }
+
+    private OCFLPersistentStorageSession createSession(final FedoraToOCFLObjectIndex index,
+                                                       final OCFLObjectSessionFactory objectSessionFactory){
+       return new OCFLPersistentStorageSession(new Random().nextLong() + "", index, objectSessionFactory);
+    }
+
```

Comment:
Two extra tests it might make sense to add now:
* test getting triples from a binary description
* test for operation that doesn't map to a persister

Reply:
+1

Reply:
Since things have changed, we should probably test:
* persist fails if commit has happened
* getTriples fails if commit has started
* commit fails if already committed
* rollback test for uncommitted session
* rollback test for session with commits to mutable heads
* we should probably test for read only session behaviors at some point (persist fail, etc)

Reply:
Will do all except the last: these behaviors are covered already by tests in OCFLPersistentSessionManagerTest.

Reply:
Okay. Also realizing we aren't testing that rollback fails if the state is not in one of the 3 happy states. Would be good to verify that